//
// PROJECT:         Aspia
// FILE:            proto/auth_session.proto
// LICENSE:         Mozilla Public License Version 2.0
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package aspia.proto.auth;

enum Method
{
    METHOD_UNKNOWN = 0;
    METHOD_BASIC   = 1;
}

enum SessionType
{
    SESSION_TYPE_UNKNOWN        = 0;
    SESSION_TYPE_DESKTOP_MANAGE = 1;
    SESSION_TYPE_DESKTOP_VIEW   = 2;
    SESSION_TYPE_POWER_MANAGE   = 4;
    SESSION_TYPE_FILE_TRANSFER  = 8;
    SESSION_TYPE_SYSTEM_INFO    = 16;
}

enum Status
{
    STATUS_UNKNOWN       = 0;
    STATUS_SUCCESS       = 1;
    STATUS_ACCESS_DENIED = 2;
}

// Host to client.
message Request
{
    Method method = 1;
    bytes nonce = 2;
}

// Client to host.
message Response
{
    SessionType session_type = 1;
    bytes key = 10;
}

// Host to client.
message Result
{
    Status status = 1;
}
